app/controllers/actions/JourneyRefiner.scala

```diff
@@ -39,18 +40,30 @@ class JourneyRefiner @Inject() (movementRepository: CacheRepository)(implicit va
     override protected def executionContext: ExecutionContext = exc
   }
 
-  private def toJourneyRequest[A](request: AuthenticatedRequest[A], types: JourneyType*): Future[Option[JourneyRequest[A]]] =
-    movementRepository.findByEori(request.user.eori).map { cacheOption =>
+  private def toJourneyRequest[A](request: AuthenticatedRequest[A], types: JourneyType*): Future[Option[JourneyRequest[A]]] = {
+    val eori = request.user.eori
+
+    movementRepository.findByEori(eori).map { cacheOption =>
       for {
         cache <- cacheOption
         answers <- cache.answers
         if types.isEmpty || types.contains(answers.`type`)
+        if isUserPermittedArriveDepartAccess(answers.`type`, eori, arriveDepartAllowList)
       } yield JourneyRequest(cache, request)
     }
+  }
 
   private def orRedirect[A](journeyOption: Option[JourneyRequest[A]]): Either[Result, JourneyRequest[A]] =
     journeyOption.toRight {
       Results.Redirect(controllers.routes.ChoiceController.displayChoiceForm())
     }
 
+  private def isUserPermittedArriveDepartAccess(answersType: JourneyType, eori: String, arriveDepartAllowList: ArriveDepartAllowList): Boolean =
+    if (Seq(ARRIVE, DEPART).contains(answersType)) arriveDepartAllowList.contains(eori)
+    else true
+}
+
+class ArriveDepartAllowList @Inject() (configuration: Configuration) {
```

Comment:
Any reason for not injecting `Configuration` directly in `JourneyRefiner` and getting rid of `ArriveDepartAllowList`?

Reply:
Where I have `ArriveDepartAllowList` being called in `ChoicePageConfig` I'd have to reimplement the logic of `ArriveDepartAllowList` in two different places if I injected `Configuration` instead.